### PR TITLE
Remove legacy container metrics collection code

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Clean up old metrics collection code: most of the code was unused since migration to `v2`, two other components were still calling the old `containers` package:
- Hostname resolution for the UTS mode check: the check has been moved to use UTS NS inode identifier, which will be more reliable and work everywhere (not only Docker)
- ECS v1 was using `GetDefaultGateway` in an attempt to find ECS Agent IP, this code has been migrated to the `util/system` package.

### Motivation

Cleanup

### Describe how to test/QA your changes

There's nothing specific to QA, but we want to make sure key features that were historically using `containers` package still work, namely:
- ECS Linux & Windows
- Docker
- Hostname resolution on Docker without Docker socket mounted and `--uts host`
- Hostname resolution on Kubernetes without Kubelet access and without CRI socket mounted

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]
Nice work 🎉
Let's just make sure we run the e2e tests on this branch before merging.

File: pkg/security/probe/activity_dump_manager.go [hunk 1]
```diff
@@ -24,9 +24,6 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/security/probe/dump"
 	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
 	"github.com/DataDog/datadog-agent/pkg/security/utils"
-
-	// util.GetHostname(...) will panic without this import
```
Comment (paulcacheux) [inline pkg/security/probe/activity_dump_manager.go:28]: Are we sure this provider is correctly initialized now ? What is the reasoning behind removing this code ?

Comment (vboulineau) [reply pkg/security/probe/activity_dump_manager.go:28]: The old provider do not exist anymore, so no need to initialize anything.
The `hostname` resolution code does not rely on the new provider either.

Comment (paulcacheux) [reply pkg/security/probe/activity_dump_manager.go:28]: Oh great then !

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/probe/activity_dump_manager.go (hunk 1 above)
Comment (paulcacheux) [reply pkg/security/probe/activity_dump_manager.go:28]: I added our team label on this PR so that we can QA it carefully since it can have some impact

**Review by lebauce — APPROVED** [review]

**Comment by djmitche** [comment]
I've taken a look at this PR a few times, and I'm really not sure how to approach it, beyond "it seems to work, let's ship it".  Any particular advice for reviewing?

**Comment by vboulineau** [comment]
> I've taken a look at this PR a few times, and I'm really not sure how to approach it, beyond "it seems to work, let's ship it". Any particular advice for reviewing?

You can discard most of the PR, i.e. all the deletions, it's dead code. The only code that still used was in `pkg/util/hostname`. It has been replaced by a better alternative (checking UTS Ns Inode number), it will **fix** many cases were the Agent was happily running with a wrong hostname.

I should made 3 separate commits and I'm sorry for that, it makes the review unnecessarily difficult and time consuming.

File: pkg/util/hostname/os_hostname_linux.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+package hostname
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/system"
+)
+
```
Comment (djmitche) [inline pkg/util/hostname/os_hostname_linux.go:17]: This seems to be the crux of things, and my searches for "linux UTS" aren't finding much on /proc/self/uts and various inode numbers.  Could you add a comment here explaining how this works and linking to relevant kernel docs?

Comment (vboulineau) [reply pkg/util/hostname/os_hostname_linux.go:17]: The explanations are present in https://github.com/DataDog/datadog-agent/blob/91a23189e9c932ad8d3fdc15012743679c86a21f/pkg/util/system/namespace_linux.go

**Review by djmitche — APPROVED** [review]
Awesome, thanks!

**Review by robertjli — APPROVED** [review]
LGTM for Processes

**Review by mellon85 — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]
LGTM for `pkg/util/cloudproviders/cloudfoundry/garden_test.go`.

**Pull request merged**